chips/sam4l/src/pm.rs

```diff
@@ -259,243 +148,170 @@ pub enum SystemClockSource {
     /// Use the internal digital frequency locked loop (DFLL) sourced from
     /// the internal RC32K clock. Note this typically requires calibration
     /// of the RC32K to have a consistent clock.
-    DfllRc32k,
+    DfllRc32kAt48MHz,
 
     /// Use an external crystal oscillator as the direct source for the
     /// system clock.
-    ExternalOscillator,
+    ExternalOscillatorAt16MHz,
 
     /// Use an external crystal oscillator as the input to the internal
     /// PLL for the system clock. This expects a 16 MHz crystal.
-    ExternalOscillatorPll,
+    PllExternalOscillatorAt48MHz,
+}
+
+/// Which frequency range does your external oscillator fall in? Configuration
```

Comment:
A more "comment-style" comment would be better: "`OscClock is the frequency range of an external oscillator..."

Why frequency range? Isn't it a specific frequency?